crates/net/eth-wire-types/src/snap.rs

```diff
@@ -208,6 +208,15 @@ pub enum SnapProtocolMessage {
     TrieNodes(TrieNodesMessage),
 }
 
+/// Decoding protocol message variants based on message ID
+macro_rules! decode_snap_message_variant {
+    ($message_id:expr, $buf:expr, $id:expr, $variant:ident, $msg_type:ty) => {
+        if $message_id == $id as u8 {
+            return Ok(Self::$variant(<$msg_type>::decode($buf)?));
+        }
+    };
+}
```

Comment:
nit, since we only need this in one location, I'd prefer if we move this macro def inside the decode fn